_health-care/_js/_components/full_name.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import _ from 'lodash';
+
+class FullName extends React.Component {
+  constructor() {
+    super();
+    this.state = {hasError: false};
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  componentWillMount() {
+    this.id = _.uniqueId('date-input-');
+  }
+
+  handleChange() {
+    const name = {
+      first: this.refs.first.value,
+      middle: this.refs.middle.value,
+      last: this.refs.last.value,
+      suffix: this.refs.suffix.value
+    }
+
+    if (!this.validate(name.first, name.last)) {
+      this.setState({hasError: true});
+    } else {
+      this.setState({hasError: false});
+    }
+
+    this.props.onUserInput(name);
+  }
+
+  validate(first, last) {
+    let valid = first !== '' && last !== '';
+    return valid;
+  }
+
+  render() {
+    const error_class = this.state.hasError ? "usa-input-error" : ""
+    return (
+      <div>
+        <div className={error_class}>
```

Comment:
Don't do this yet...but I'm wondering if we need a TextInput component that takes in a `allowedCharacters` and `isRequired` prop.